File: client-app/src/app/models/session.ts
import { Attendee } from "./attendance";

export interface SessionFormValues {
    sessionName: string
    sessionExpiresAt: string
    linkExpiryFreequency: number
    regenerateLinkToken: boolean
}

export interface Session {
    sessionId: string
    sessionName: string
    sessionExpiresAt: string
    hostName: string
    linkToken: string
    attendeesCount: number
    status: string
    linkExpiryFreequency: number
    regenerateLinkToken: boolean
}

export interface SessionAttendees {
    sessionId: string
    sessionName: string
    sessionExpiresAt: string
    hostName: string
    linkToken: string
    attendees: Attendee[];
    attendeesCount: number
    status: string
}

File: API/Entities/Session.cs
using System.ComponentModel.DataAnnotations;
using API.Enums;

namespace API.Entities;
public class Session
{
    public Guid Id { get; set; }
    public string HostId { get; set; }
    public AppUser Host { get; set; }
    [Required]
    [MaxLength(50)]
    [MinLength(3)]
    public string SessionName { get; set; }
    [Required]
    public DateTime SessionExpiresAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public bool RegenerateLinkToken { get; set; } = true;
    public int LinkExpiryFreequency { get; set; } = 30;
    public List<Attendee> Attendees { get; set; } = new List<Attendee>();

    public List<RefereshLinkToken> RefereshLinkTokens { get; set; } = new List<RefereshLinkToken>();

    public int AttendeesCount { get { return Attendees.Count; } }

    public string Status { get { return SessionExpiresAt > DateTime.UtcNow ? SessionStatus.Active.ToString() : SessionStatus.Expired.ToString(); } }
}